db/seeds.rb

```diff
@@ -1,7 +1,79 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+require 'csv'
+
+contestants = [
+  {
+    name:  "Pavan",
+    email: "pavan.sarguru@gmail.com"
+  },
+  {
+    name:  "Logan",
+    email: "logandsprice@gmail.com"
+  },
+  {
+    name:  "Peter",
+    email: "peterbutzen@gmail.com"
+  },
+  {
+    name:  "Samarth",
+    email: "samarth.bhaskar@gmail.com"
+  },
+  {
+    name:  "Piku",
+    email: "pulkitgoel92@gmail.com"
+  },
+  {
+    name:  "Neal",
+    email: "ndodia2@gmail.com"
+  }
+]
+
+Contestant.destroy_all
+contestants.each do |c|
+  contestant = Contestant.new(
+    name:  c[:name],
+    email: c[:email]
+  )
+  contestant.save if contestant.valid?
+end
+
+url = "http://data.nba.net/10s/prod/v2/2018/teams.json"
+response = HTTParty.get(url)
+parsed = response.parsed_response
+nba_teams = parsed["league"]["standard"].select {|t| t["isNBAFranchise"] == true }
+
+# team_file = File.read("teams.json")
+# teams_json = JSON.parse(team_file)
+# nba_teams = teams_json["league"]["standard"].select {|t| t["isNBAFranchise"] == true }
+
+Team.destroy_all
+nba_teams.each do |t|
+  team =  Team.new(
+    nba_id: t["teamId"],
+    city:   t["city"],
+    name:   t["nickname"],
+    conference: t["confName"],
+    division:   t["divName"]
+  )
+  team.save if team.valid?
+end
+
+Pick.destroy_all
+locks = {
+  "Pavan" => ["Raptors", "Bulls", "Rockets"],
+  "Logan" => ["Celtics", "Bucks", "Bulls"],
+  "Peter" => ["76ers", "Rockets", "Kings"],
+  "Samarth" => ["Bucks", "Lakers", "Timberwolves"],
+  "Piku" => ["Cavaliers", "Timberwolves", "Grizzlies"],
+  "Neal" => ["Magic", "Jazz", "Nuggets"]
+}
+CSV.foreach("picks.csv", headers: true) do |row|
+  team = Team.find_by_name(row["Team"])
+  team.update_attributes(line: row["Win Total"].to_f)
+
+  row.headers.each do |h|
+    if c = Contestant.find_by(name: h)
+      team.picks.create(contestant: c, over: row[h] == "OVER", lock: team.name.in?(locks[c.name]))
+    end
```

Comment:
I'm surprised this works.  I would think that `row.headers.each` would loop through each header including 'Team' and 'Win Total'.  Then I could think when you do `Contestant.find_by(name: h)`, it would throw an exception when it couldn't find a contestant with name "Team".

Reply:
I leveraged how `find_by` returns `nil` if the search fails whereas `find` throws an exception.  I think the assumption is if you use `find` and submit an id, it's assumed you know exactly what you want so it makes sense to throw the exception, as opposed to searching for something with `find_by` that may or may not exist.